paginator/paginator.go

```diff
@@ -133,8 +133,15 @@ func (p *Paginator) setup(db *gorm.DB, dest interface{}) error {
 			sqlKey := p.parseSQLKey(db, dest, rule.Key)
 			rule.SQLRepr = fmt.Sprintf("%s.%s", sqlTable, sqlKey)
 		}
+
 		if rule.NULLReplacement != nil {
-			rule.SQLRepr = fmt.Sprintf("COALESCE(%s, '%v')", rule.SQLRepr, rule.NULLReplacement)
+			var nullReplacement string
+			if rule.SQLType != nil {
+				nullReplacement = fmt.Sprintf("CAST('%v' as %s)", rule.NULLReplacement, *rule.SQLType)
+			} else {
+				nullReplacement = fmt.Sprintf("'%v'", rule.NULLReplacement)
+			}
+			rule.SQLRepr = fmt.Sprintf("COALESCE(%s, %s)", rule.SQLRepr, nullReplacement)
```

Comment:
nit: one style suggestion. Does this also work?
```suggestion
			nullReplacement := fmt.Sprintf("'%v'", rule.NULLReplacement)
			if rule.SQLType != nil {
				nullReplacement = fmt.Sprintf("CAST(%s as %s)", nullReplacement, *rule.SQLType)
			}
			rule.SQLRepr = fmt.Sprintf("COALESCE(%s, %s)", rule.SQLRepr, nullReplacement)
```

Reply:
Makes sense

Reply:
Thanks!